src/tools/compiletest/src/runtest.rs

```diff
@@ -4014,9 +4014,17 @@ impl<'test> TestCx<'test> {
         explicit_format: bool,
     ) -> usize {
         let stderr_bits = format!("{}bit.stderr", self.config.get_pointer_width());
+        let force_color_svg = self.props.compile_flags.iter().any(|s| s.contains("--color"))
+            && !self.config.target.contains("windows");
```

Comment:
Why is this necessary?

Reply:
It's not absolutely necessary, but I don't have access to a windows box to ensure this test ran *and* the colors for the ANSI escapes is subtly different between windows and non-windows. Can change that and do a full build on CI to get the windows svg file if we want to land it with it.

Reply:
For annotate-snippets, @Muscraft  recently added a `testing-colors` feature flag to force the colors the same across all platforms so we don't have to deal with platform-specific colors (except for the places we need to to verify they are platform-specific)

Reply:
I'm just worried that when ui testing on windows platforms, this will cause test failures since compiletest expects stdout files but there are svg files here.

Reply:
I *believe* that the filter I included was for *explicit* `windows` platform request from the test file. I will remove the platform check.

@epage is `testing-colors` included in 0.1.3?

Reply:
`testing-colors` is a annotate-snippets feature.  Not really relevant until rustc is converted over but just giving an idea of strategies.

Reply:
> I believe that the filter I included was for explicit windows platform request from the test file. I will remove the platform check.

For the record, the variable `self.config.target` is set for all UI tests, so I believe this will produce observable behavioral differences in compiletest, and not just for tests tagged `// only-windows`/`// ignore-windows`.

> I will remove the platform check.

Cool, that sounds like the easiest solution.

Reply:
Another solution to this might be to pull the colors out into a separate crate and share them between `rustc` and `annotate-snippets`. I have thought about doing this before, but have never talked with enough people to see if it was a good idea. It would make it easier to share the colors during the transition period and allow other crates to match colors in the long run.